deployment/ccip/changeset/initial_add_chain.go

```diff
@@ -41,67 +39,42 @@ func ConfigureNewChains(env deployment.Environment, c NewChainsConfig) (deployme
 	}, nil
 }
 
-type USDCConfig struct {
-	EnabledChains []uint64
-	USDCAttestationConfig
-	CCTPTokenConfig map[ccipocr3.ChainSelector]pluginconfig.USDCCCTPTokenConfig
-}
-
-func (cfg USDCConfig) EnabledChainMap() map[uint64]bool {
-	m := make(map[uint64]bool)
-	for _, chain := range cfg.EnabledChains {
-		m[chain] = true
-	}
-	return m
-}
-
-func (cfg USDCConfig) ToTokenDataObserverConfig() []pluginconfig.TokenDataObserverConfig {
-	return []pluginconfig.TokenDataObserverConfig{{
-		Type:    pluginconfig.USDCCCTPHandlerType,
-		Version: "1.0",
-		USDCCCTPObserverConfig: &pluginconfig.USDCCCTPObserverConfig{
-			Tokens:                 cfg.CCTPTokenConfig,
-			AttestationAPI:         cfg.API,
-			AttestationAPITimeout:  cfg.APITimeout,
-			AttestationAPIInterval: cfg.APIInterval,
-		},
-	}}
-}
-
-type USDCAttestationConfig struct {
-	API         string
-	APITimeout  *config.Duration
-	APIInterval *config.Duration
-}
-
 type CCIPOCRParams struct {
-	OCRParameters         types.OCRParameters
-	CommitOffChainConfig  pluginconfig.CommitOffchainConfig
+	OCRParameters types.OCRParameters
+	// Note contains pointers to Arb feeds for prices
+	CommitOffChainConfig pluginconfig.CommitOffchainConfig
+	// Note ontains USDC config
 	ExecuteOffChainConfig pluginconfig.ExecuteOffchainConfig
 }
 
-func (p CCIPOCRParams) Validate() error {
-	if err := p.OCRParameters.Validate(); err != nil {
+func (c CCIPOCRParams) Validate() error {
+	if err := c.OCRParameters.Validate(); err != nil {
 		return fmt.Errorf("invalid OCR parameters: %w", err)
 	}
-	if err := p.CommitOffChainConfig.Validate(); err != nil {
+	if err := c.CommitOffChainConfig.Validate(); err != nil {
 		return fmt.Errorf("invalid commit off-chain config: %w", err)
 	}
-	if err := p.ExecuteOffChainConfig.Validate(); err != nil {
+	if err := c.ExecuteOffChainConfig.Validate(); err != nil {
 		return fmt.Errorf("invalid execute off-chain config: %w", err)
 	}
 	return nil
 }
 
 type NewChainsConfig struct {
-	HomeChainSel   uint64
-	FeedChainSel   uint64
-	ChainsToDeploy []uint64
-	TokenConfig    TokenConfig
-	USDCConfig     USDCConfig
-	// For setting OCR configuration
-	OCRSecrets deployment.OCRSecrets
-	OCRParams  map[uint64]CCIPOCRParams
+	// Common to all chains
```

Comment:
this is the main change. We separate into global and per chain, and use the Exec/Commit plugin config directly to reduce some duplication